### PR TITLE
add asech to evaluate class

### PR DESCRIPTION
**Pull request by ranjithkumar007**
fix for #1188 @jppelteret

### CODE REVIEW

File: symengine/real_mpfr.cpp [hunk 1]
```diff
@@ -964,6 +964,27 @@ class EvaluateMPFR : public Evaluate
             "Result is complex. Recompile with MPC support.");
 #endif
     }
+    virtual RCP<const Basic> asech(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<RealMPFR>(x))
+        mpfr_srcptr x_ = down_cast<const RealMPFR &>(x).i.get_mpfr_t();
+        if (mpfr_cmp_si(x_, 0) >= 0 or mpfr_cmp_si(x_, 1) <= 0) {
```
Comment (isuruf) [inline symengine/real_mpfr.cpp]: I think `or` should be changed to `and`

**Comment by jppelteret** [comment]
Thanks for the quick fix @ranjithkumar007 !

File: symengine/real_mpfr.cpp (hunk 1 above)
Comment (ranjithkumar007) [inline symengine/real_mpfr.cpp]: yes..thanks for the catch

**Comment by ranjithkumar007** [comment]
@isuruf  i think get_eval should also be added here before adding any testcase [asech](https://github.com/symengine/symengine/blob/master/symengine/functions.cpp#L2005-L2011)

**Comment by ranjithkumar007** [comment]
Ping @isuruf

**Comment by jppelteret** [comment]
Looks pretty good @ranjithkumar007 :-) Thanks for your work on this and the comprehensive test cases!

**Comment by ranjithkumar007** [comment]
@isuruf @jppelteret  tests have passed and finally [codecov](https://codecov.io/gh/ranjithkumar007/symengine/branch/asech) shows coverage at 71.03%(master is at 70.95) ..

**Pull request merged**